src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -326,10 +326,19 @@ private ValueTask<HttpConnection> GetOrReserveHttp11ConnectionAsync(Cancellation
 
             // See if we have an HTTP2 connection
             Http2Connection http2Connection = _http2Connection;
+            TimeSpan pooledConnectionLifetime = _poolManager.Settings._pooledConnectionLifetime;
+
+            // Check only lifetime. There is no need to check idle if we are about to use it.
+            if (http2Connection != null && http2Connection.IsExpired(DateTimeOffset.UtcNow, pooledConnectionLifetime, Timeout.InfiniteTimeSpan))
```

Comment:
Can we move all of this down below under the "if (http2Connection != null)" check?